src/plugins/data/public/query/saved_query/saved_query_service.ts

```diff
@@ -19,7 +24,8 @@ type SerializedSavedQueryAttributes = SavedObjectAttributes &
   };
 
 export const createSavedQueryService = (
-  savedObjectsClient: SavedObjectsClientContract
+  savedObjectsClient: SavedObjectsClientContract,
+  filterManager: FilterManager
```

Comment:
instead of providing filterManager contract here, you could import filter manager extract/inject utilities (as they are stateless)

`import { extract, inject } from '../../common/query/persistable_state';`